### PR TITLE
Updated Test Project

### LINKED ISSUE #54: Missing tests
> Good. Once it is merged we should have corresponding tests in symfony-demo 
_Originally posted by @DavertMik in https://github.com/Codeception/module-symfony/issues/50#issuecomment-732076031_

Due to the current test coverage of the module and the time it would take to correct compatibility issues in current test projects, I decided to create a [new test project](https://github.com/TavoNiievez/codeception-symfony-tests) with 3 branches:
- `main`, for the latest version of Symfony (5.2)
- `4.4`, for the latest LTS and,
- `3.4` for the oldest LTS.

<hr/>

<details>
<summary><b>Click to see progress on Symfony 5.2</b></summary>
<br/>

- [x] amLoggedInAs
- [x] amOnAction
- [x] amOnRoute
- [x] dontSeeAuthentication
- [x] dontSeeEmailIsSent
- [x] dontSeeFormErrors
- [x] grabParameter
- [x] grabRepository
- [x] grabService
- [x] logout
- [x] runSymfonyConsoleCommand
- [x] seeAuthentication
- [x] seeCurrentActionIs
- [x] seeCurrentRouteIs
- [x] seeEmailIsSent
- [x] seeFormErrorMessage
- [x] seeFormHasErrors
- [x] seeInCurrentRoute
- [x] seeInSession
- [x] seeNumRecords
- [x] seePageIsAvailable
- [x] seePageRedirectsTo
- [x] seeUserHasRole
- [x] seeUserPasswordDoesNotNeedRehash
- [x] submitSymfonyForm
</details>

<hr/>

<details>
<summary><b>Click to see progress on Symfony 4.4</b></summary>
<br/>

- [x] amLoggedInAs
- [x] amOnAction
- [x] amOnRoute
- [x] dontSeeAuthentication
- [x] dontSeeEmailIsSent
- [x] dontSeeFormErrors
- [x] grabParameter
- [x] grabRepository
- [x] grabService
- [x] logout
- [x] runSymfonyConsoleCommand
- [x] seeAuthentication
- [x] seeCurrentActionIs
- [x] seeCurrentRouteIs
- [x] seeEmailIsSent
- [x] seeFormErrorMessage
- [x] seeFormHasErrors
- [x] seeInCurrentRoute
- [x] seeInSession
- [x] seeNumRecords
- [x] seePageIsAvailable
- [x] seePageRedirectsTo
- [x] seeUserHasRole
- [x] seeUserPasswordDoesNotNeedRehash
- [x] submitSymfonyForm
</details>

<hr/>

<details>
<summary><b>Click to see progress on Symfony 3.4</b></summary>
<br/>

- [x] amLoggedInAs
- [x] amOnAction
- [x] amOnRoute
- [x] dontSeeAuthentication
- [ ] dontSeeEmailIsSent
- [x] dontSeeFormErrors
- [ ] grabParameter
- [x] grabRepository
- [x] grabService
- [x] logout
- [x] runSymfonyConsoleCommand
- [x] seeAuthentication
- [x] seeCurrentActionIs
- [x] seeCurrentRouteIs
- [ ] seeEmailIsSent
- [x] seeFormErrorMessage
- [x] seeFormHasErrors
- [x] seeInCurrentRoute
- [x] seeInSession
- [x] seeNumRecords
- [x] seePageIsAvailable
- [x] seePageRedirectsTo
- [x] seeUserHasRole
- [ ] seeUserPasswordDoesNotNeedRehash
- [x] submitSymfonyForm
</details>

<hr/>

This new test project serves as a replacement for:
[[Naktibalda/codeception-symfony-tests](https://github.com/Naktibalda/codeception-symfony-tests)] [[Codeception/symfony-demo](https://github.com/Codeception/symfony-demo)].

### PR DESCRIPTION
**Pull request by TavoNiievez**
Closes #54

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -6,106 +6,87 @@ jobs:
...
-            symfony: 4
-          - php: 7.0
-            symfony: 5
           - php: 7.1
             symfony: 5
 
     steps:
     - name: Checkout code
-      uses: actions/checkout@v2
+      uses: actions/checkout@v2.3.4
```
Comment (eXorus) [inline .github/workflows/main.yml:19]: I don't think it's useful to put the exact version.
If you let v2, you will take advantages of new version.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -6,106 +6,87 @@ jobs:
...
         coverage: none
 
     - name: Checkout Symfony 3.4 Sample
       if: matrix.symfony == 3.4
-      uses: actions/checkout@v2
+      uses: actions/checkout@v2.3.4
       with:
-        repository: Naktibalda/codeception-symfony-tests
-        path: framework-tests
+        repository: TavoNiievez/codeception-symfony-tests
```
Comment (eXorus) [inline .github/workflows/main.yml:33]: I think we should have this repository in our organization.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -6,106 +6,87 @@ jobs:
...
-        ref: symfony5
+        repository: TavoNiievez/codeception-symfony-tests
+        submodules: recursive
+        ref: master
 
     - name: Install Symfony Sample on PHP7
       if: matrix.php < 8
-      run: composer update --no-dev --prefer-dist --no-interaction
-      working-directory: framework-tests
+      run: composer update --no-progress
```
Comment (eXorus) [inline .github/workflows/main.yml:55]: Why did you remove --no-dev, --prefer-dist and --no-interaction ?

**Review by eXorus — APPROVED** [review]
I only did the migration from Travis to Github Actions.
I don't use SF or this module.

But it looks you did a great job. Thanks.
I added few comments, feel free to change or not.

File: .github/workflows/main.yml (hunk 2 above)
Comment (TavoNiievez) [reply .github/workflows/main.yml:33]: totally agree! I already talked to @DavertMik  about it a few days ago, I'm waiting for him to create the repository and give me the permissions ;-)

File: .github/workflows/main.yml (hunk 1 above)
Comment (TavoNiievez) [reply .github/workflows/main.yml:19]: [I relied on symfony/demo](https://github.com/symfony/demo/blob/d8d09068eba6bd2ad68ae6bffd88f65dc263dc21/.github/workflows/ci.yaml#L28) for this change, but using the latest available version sounds better to me too. I'll leave it as it was, thanks.

File: .github/workflows/main.yml (hunk 3 above)
Comment (TavoNiievez) [reply .github/workflows/main.yml:55]: - `no-dev`: I want the latest versions of Codeception and its modules (which is included in the `require-dev` section in the `composer.json` files) to be used when running CI jobs.

- `--no-interaction`: There will be no interaction anyway, the .lock files keep track of the installed recipes, it is very unlikely that any bundle will add a new recipe within a minor version and this also damages the continuous integration flow.

- `--prefer-dist`: Well, this [is already included](https://github.com/TavoNiievez/codeception-symfony-tests/blob/bd5b3afdb80f5bcb022ce8ee05923d237f58a2bf/composer.json#L51) in the composer.json files. In any case what the composer documentation says is *'This can speed up installs substantially on build servers and other use cases where you typically do not run updates of the vendors .'* and i am intentionally running a `composer update`.

Here I also took into account how it is done in [symfony/demo](https://github.com/symfony/demo/blob/d8d09068eba6bd2ad68ae6bffd88f65dc263dc21/.github/workflows/ci.yaml#L56), but when analyzing each individual case, I came to the conclusion that the only thing I don't want is to see the progress of locking/downloading/installation in the logs.
